plugins/net_plugin/net_plugin.cpp

```diff
@@ -1243,7 +1242,7 @@ namespace eosio {
             try {
                c->buffer_queue.clear_out_queue();
                // May have closed connection and cleared buffer_queue
-               if( !c->socket_is_open() || socket != c->socket ) {
+               if( !c->socket->is_open() || socket != c->socket ) {
                   peer_ilog( c, "async write socket ${r} before callback", ("r", c->socket_is_open() ? "changed" : "closed") );
                   c->close();
```

Comment:
Shouldn't we close `socket` here, instead of `c->socket`? as in calling `socket->close(ec);`.